docs/source/design/threaded_runner.rst

```diff
@@ -0,0 +1,87 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_threaded_runner:
+
+
+****************
+Threaded runner
+****************
+
```

Comment:
Wonderful! I was hoping for that to be in the diff, after having parser the associated code ;-)